cdap-runtime-ext-emr/src/main/java/co/cask/cdap/runtime/spi/provisioner/emr/EMRConf.java

```diff
@@ -0,0 +1,186 @@
+                  String additionalMasterSecurityGroup, String serviceRole, String jobFlowRole, String releaseLabel,
+                  String masterInstanceType, String workerInstanceType, int instanceCount,
+                  @Nullable String logURI, @Nullable SSHPublicKey publicKey) {
+    this.accessKey = accessKey;
+    this.secretKey = secretKey;
+    this.region = region;
+    this.ec2SubnetId = ec2SubnetId;
+    this.additionalMasterSecurityGroup = additionalMasterSecurityGroup;
+    this.serviceRole = serviceRole;
+    this.jobFlowRole = jobFlowRole;
+    this.releaseLabel = releaseLabel;
+    this.logURI = logURI;
+    this.masterInstanceType = masterInstanceType;
+    this.workerInstanceType = workerInstanceType;
+    this.instanceCount = instanceCount;
+    this.publicKey = publicKey;
+  }
+
+  String getRegion() {
+    return region;
+  }
+
+  String getEc2SubnetId() {
+    return ec2SubnetId;
+  }
+
+  String getAdditionalMasterSecurityGroup() {
+    return additionalMasterSecurityGroup;
+  }
+
+  String getServiceRole() {
+    return serviceRole;
+  }
+
+  String getJobFlowRole() {
+    return jobFlowRole;
+  }
+
+  String getReleaseLabel() {
+    return releaseLabel;
+  }
+
+  int getInstanceCount() {
+    return instanceCount;
+  }
+
+  String getMasterInstanceType() {
+    return masterInstanceType;
+  }
+
+  String getWorkerInstanceType() {
+    return workerInstanceType;
+  }
+
+  @Nullable
+  String getLogURI() {
+    return logURI;
+  }
+
+  @Nullable
+  SSHPublicKey getPublicKey() {
+    return publicKey;
+  }
+
+  AWSCredentialsProvider getCredentialsProvider() {
+    return new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secretKey));
+  }
+
+  static EMRConf fromProvisionerContext(ProvisionerContext context) {
+    Optional<SSHKeyPair> sshKeyPair = context.getSSHContext().getSSHKeyPair();
+    return create(context.getProperties(), sshKeyPair.map(SSHKeyPair::getPublicKey).orElse(null));
+  }
+
+  /**
+   * Create the conf from a property map while also performing validation.
+   */
+  static EMRConf fromProperties(Map<String, String> properties) {
+    return create(properties, null);
+  }
+
+  private static EMRConf create(Map<String, String> properties, @Nullable SSHPublicKey publicKey) {
+    String accessKey = getString(properties, "accessKey");
+    String secretKey = getString(properties, "secretKey");
+
+    String region = getString(properties, "region", Regions.DEFAULT_REGION.getName());
+
+    String ec2SubnetId = getString(properties, "ec2SubnetId");
+    String additionalMasterSecurityGroup = getString(properties, "additionalMasterSecurityGroup");
+    String serviceRole = getString(properties, "serviceRole", "EMR_DefaultRole");
+    String jobFlowRole = getString(properties, "jobFlowRole", "EMR_EC2_DefaultRole");
+
+    // all 4.9.x is java 7... which we don't support, so EMR 5.0.0 is our minimum
+    String releaseLabel = getString(properties, "releaseLabel", "emr-5.15.0");
```

Comment:
as discussed, don't think we should allow users to set the version in the first cut until we have done more extensive testing.